### PR TITLE
ci: lint dagger using dagger

### PR DESCRIPTION
**Pull request by aluzzardi**
- `Makefile` now uses `dagger do lint` to lint
- GHA uses dagger to lint

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,23 +33,24 @@ test: # Run all tests
...
 
 .PHONY: golint
-golint: # Go lint
-	golangci-lint run --timeout 3m
+golint: dagger # Go lint
+	./cmd/dagger/dagger do lint go
 
 .PHONY: cuefmt
-cuefmt: # Format all cue files
+cuefmt: dagger # Format all cue files
```
Comment (TomChv) [inline Makefile:40]: That command do not require dagger to works

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -41,35 +41,22 @@ jobs:
...
-          make cuelint
+      - name: "Set up dagger"
+        uses: dagger/dagger-for-github@v2
+        with:
+          install-only: true
 
-      - name: "Lint docs"
+      - name: Lint
+        env:
+          DAGGER_CACHE_TO: "type=gha,mode=max,scope=dagger-ci-lint"
```
Comment (TomChv) [inline .github/workflows/lint.yml:59]: Why not `DAGGER_LOG_FORMAT: plain` here?

**Review by TomChv — CHANGES_REQUESTED** [review]
LGTM, but I saw that CI has timeout, it's strange because we can see that operation are `CACHED`.

**Comment by aluzzardi** [comment]
Thanks @TomChv 

I'm not 100% sure if the cache is working correctly, could you take a look?

**Comment by aluzzardi** [comment]
@TomChv Ok here's my theory: As soon as we have a `#CacheDir`, since they're not persisted, it invalidates the cache resulting in a re-run.

Basically everything with a `#CacheDir` is executed every single time.

Is that a good assumption?

**Comment by aluzzardi** [comment]
> I'm not sure because I could make cache works with go.#Build and it's using core.#CacheDir in go.#Container.

The `Dagger CI / Build` job here is using `go.#Build` and is not getting cached. I just re-triggered it 4 times.

**Comment by TomChv** [comment]
> The Dagger CI / Build job here is using go.#Build and is not getting cached. I just re-triggered it 4 times.

Ok that's weird, in `Dagger CI / Build`, some action are cached but I see that the [huge one isn't](https://github.com/dagger/dagger/runs/5861154039?check_suite_focus=true#step:6:917).
Cache is definitely working but only in specific context. I must find what's going wrong with that.

**Pull request merged**